src/java/org/apache/cassandra/tools/nodetool/ListEndpointsPendingHints.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.tools.nodetool;
+
+import java.net.UnknownHostException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import io.airlift.airline.Command;
+import org.apache.cassandra.hints.PendingHintsInfo;
+import org.apache.cassandra.locator.EndpointSnitchInfoMBean;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.tools.NodeProbe;
+import org.apache.cassandra.tools.NodeTool;
+import org.apache.cassandra.tools.nodetool.formatter.TableBuilder;
+
+@Command(name = "listendpointspendinghints", description = "Print all the endpoints that this node has hints for")
+public class ListEndpointsPendingHints extends NodeTool.NodeToolCmd
+{
+    @Override
+    public void execute(NodeProbe probe)
+    {
+        List<PendingHintsInfo> pendingHints = probe.listPendingHints();
+        if(pendingHints.isEmpty())
+        {
+            probe.output().out.println("This node does not have hints for other endpoints");
+        }
+        else
+        {
+            Map<String, String> endpointMap = probe.getHostIdToEndpointWithPort();
+            Map<String, String> simpleStates = probe.getSimpleStatesWithPort();
+            EndpointSnitchInfoMBean epSnitchInfo = probe.getEndpointSnitchInfoProxy();
+
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss,SSS");
```

Comment:
I haven't seen comma delimiter for milliseconds previously, usually it is like "yyyy-MM-dd HH:mm:ss.SSS" (dot instead of comma). Check examples at https://docs.oracle.com/javase/6/docs/api/java/text/SimpleDateFormat.html.

What timezone the dates will be printed at? I feel we need to print the timezone ("X" pattern) itself and also we need to decide whether we want to see local or UTC timezone in the output. For me UTC makes more sense because theoretically nodes may have different zones and the mesh (as Brandon mentioned in the ticket) may take the output from many servers.

nit: I'd suggest to use `DateTimeFormatter` and modern Java 8 classes, they seem to be more convenient to use and they force a user to to raise timezone questions (unlikely to to old class that silently rely on the local timezone).

Reply:
I don't really know how TimestampType columns int the VT's output would look like, but probably it makes sense to use the same format/timezone here. Just for unification purpose.

Reply:
Actually it seems the comma separation is used project-wise. I just did a quick grep. You can see in RepairRunner, BootstrapMonitor and NodeTool. But we have it also with a dot in TimestampSerializer.

Reply:
hah, looks weird! I see no reasons for using such a weird format, but it is good to know!

Reply:
@azotcsit That I don't argue about but it seems it's been in nodetool like this since 2014... 
I love using `grep` and `git blame`  :D

Reply:
Even though I also find this datetime format... unconventional, I think consistency in nodetool commands' output is more important. In my view, all commands should use the same format for datetime. If we want to change this format, then we should open a separate issue for that.

As of now, the output looks like the following:

```
✗ ccm node1 nodetool listpendinghints

Host ID                              Address        Rack  DC          Status Total files Newest                  Oldest
26a8a1cc-fcb4-465b-953f-eaec936b8b04 127.0.0.2:7000 rack1 datacenter1 DOWN   6           2021-10-02 12:36:57,801 2021-10-02 12:23:01,923

✗ ccm node1 cqlsh

cqlsh> select * from system_views.pending_hints ;

 host_id                              | address   | dc          | files | newest                          | oldest                          | port | rack  | status
--------------------------------------+-----------+-------------+-------+---------------------------------+---------------------------------+------+-------+--------
 26a8a1cc-fcb4-465b-953f-eaec936b8b04 | 127.0.0.2 | datacenter1 |     2 | 2021-10-02 12:36:42.005000+0000 | 2021-10-02 12:23:01.923000+0000 | 7000 | rack1 |   DOWN

(1 rows)
```

I also replaced `SimpleDateFormat` with `DateTimeFormatter` in af9e91019775d70871f112e2807b91bec44c39d5.

Reply:
Thank for showing sample outputs, it is very useful!

nit: does it make sense to have this output in the same order as the VT has?

Reply:
> does it make sense to have this output in the same order as the VT has?

They were supposed to be in the same order :(

https://github.com/apache/cassandra/blob/af9e91019775d70871f112e2807b91bec44c39d5/src/java/org/apache/cassandra/db/virtual/PendingHintsTable.java#L55-L71

https://github.com/apache/cassandra/blob/af9e91019775d70871f112e2807b91bec44c39d5/src/java/org/apache/cassandra/tools/nodetool/ListPendingHints.java#L57

AFAIU Cassandra sorts regular columns in a natural order, not sure we can do anything about it here:

https://github.com/apache/cassandra/blob/af9e91019775d70871f112e2807b91bec44c39d5/src/java/org/apache/cassandra/db/RegularAndStaticColumns.java#L164-L170

Reply:
yeah, I just recently discovered that "sorting" logic too: https://issues.apache.org/jira/browse/CASSANDRA-16976?focusedCommentId=17422708&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17422708

So I feel it makes sense to change the order in nodetool to match the VT as I did in CASSANDRA-16976, but I do not think it is a strict requirement, so I'd see what @ekaterinadimitrova2 says.